erpnext/assets/doctype/asset/depreciation.py

```diff
@@ -740,6 +740,12 @@ def get_disposal_account_and_cost_center(company):
 def get_value_after_depreciation_on_disposal_date(asset, disposal_date, finance_book=None):
 	asset_doc = frappe.get_doc("Asset", asset)
 
+	if str(asset_doc.available_for_use_date) > disposal_date:
+		frappe.throw("Posting Date of asset capitalization should be after 'Available-for-use Date' of asset.")
+
+	if str(asset_doc.available_for_use_date) == disposal_date:
```

Comment:
Instead of using `str()`, it's better to use `getdate(disposal_date)` for the comparison. Also, use `elif` here.